sources/core/Stride.Core/Utilities.cs

```diff
@@ -111,37 +111,6 @@ public static unsafe void FreeMemory(nint alignedBuffer)
             }
         }
 
-        /// <summary>
-        ///   Read stream to a byte[] buffer
-        /// </summary>
-        /// <param name = "stream">input stream</param>
-        /// <returns>a byte[] buffer</returns>
-        [Obsolete("Allocates. Read into the destination.")]
```

Comment:
The reason this method was marked obsolete was that it allocates a buffer for copying the stream data into and it was preferable a variant that copies directly to a provided buffer.
Your alternative using `MemoryStream` does the same.
Would not be better to provide the alternative that can read a `Stream` directly to a `Span<byte>`?

Reply:
i will check that out

Reply:
what about that? i dont really get how i can convert a span to a byte[] else
```csharp
    public class DataContainer 
    {
        public byte[] Data;

        public static DataContainer Load(Stream stream)
        {
            byte[] dataArray = new byte[stream.Length];
            stream.Read(dataArray);
            return new DataContainer { Data = dataArray };
        }
```
if i read directly to the span it would look like that ( i dont have that much expierence with spans )
```csharp
        public byte[] Data;

        public static DataContainer Load(Stream stream)
        {
            Span<byte> buffer = new Span<byte>();
            stream.Read(buffer);
            return new DataContainer { Data = buffer.ToArray() };
        }
```
this would again including copying the content according to ToArray docs of span

Reply:
Using spans only makes sense if you're working within the stack for the operations. Here the class uses an array in memory for long term access which means using the array is the valid approach.

Reply:
will apply that

Reply:
https://github.com/stride3d/stride/pull/1675/commits/6df7f021f3fb9ad07a3841462361354eee931de5
done with that